### ISSUE TITLE
PFAM and external database (CAZY) annotations are missing in orftable and sqmtables

### ISSUE THREAD

**Issue by chandnisidhu** [jtamames/SqueezeMeta#280]
Hi,

I used external gene calls (FragGeneScan) in SqueezeMeta pipeline. My pipeline has been completed successfully but the results were different as compared to default Prodigal pipeline. I found that PFAM and CAZY annotations are missing from final sqm tables and also from orftable and contigtable. But information about them is present in respective fun files in results folder. 
Other information regarding gene length, GC content is also missing. Is this because of using FragGeneScan gff files?
I am attaching my log files along with some other parameters files.

Thank you in advance 

Chandni

[Files.zip](https://github.com/jtamames/SqueezeMeta/files/6298070/Files.zip)

**Comment by fpusan** [comment]
Hi,
If they are missing from the `13.*.orftable` file then it's normal that they are missing from the final tables. `sqm2tables.py` is creating the `PFAM` and `CAZy` files since `SqueezeMeta_conf.pl` and the fun files indicates that there should be `PFAM` and `CAZy` results. But since they are not actually in the `orftable` it doesn't add something.
What is the restart history of your project? Maybe the fun files are referring to genes present in the old gff files?

**Comment by chandnisidhu** [comment]
Hi,

No I double-checked it. They are not from the old gff files. 

I first did the whole pipeline with default tools and found out that some genes are kind of truncated so I did ORF prediction with FragGeneScan and replaced the .fna, .faa and .gff files. After that, I restarted the pipeline from step 4.

**Comment by jtamames** [comment]
I don´t think it is related to the gff files, becase you say that the rest of annotations (taxa, COGs, KEGG) are correct, aren´t they? Then the genes have been called properly, and the homology searches and mapping are working. Could you `tree` your project to see all the files and the sizes? Please do `tree -h -D <projectname>`. The tree command is installed in the SqueezeMeta distribution.

Best,
Javier

**Comment by chandnisidhu** [comment]
Also, the contigstable file is different. In my previous results, each line in this file has information about a single contig. Now it contains ORF information in each line.

**Comment by jtamames** [comment]
Ok, that means that the genes are being treated as contigs. That could happen if it ran prodigal on your FragScan predictions... does that make sense?

**Comment by chandnisidhu** [comment]
But why would it do it? I restarted it from step 4.

**Comment by jtamames** [comment]
And you kept the original 01.<project>.fasta file?

**Comment by chandnisidhu** [comment]
Yes

**Comment by jtamames** [comment]
Could you please send me the 01 and 03 files by wetransfer, so that I can take a look?

**Comment by chandnisidhu** [comment]
By any chance, would it be because of blastx? (step 8)

Because I just had a look on stats file, number of orphans(no hits) is more than the predicted ORFs which does not make any sense.

**Comment by jtamames** [comment]
Clearly something is wrong at some point... I don´t think is because step 08, but it could be. Then please send me also the `08.<project>.gff`
Thanks

**Comment by chandnisidhu** [comment]
Okay I am sending you the required files.

Best,
Chandni

**Comment by chandnisidhu** [comment]
Hi,

This is to confirm if you have received my files.

Best,

Chandni

**Comment by jtamames** [comment]
Yes, thank you, I will let you know asap

**Comment by jtamames** [comment]
Ok. This is exactly the same error that you posted in #280. So the solution is the same.

**Comment by chandnisidhu** [comment]
Hi,

Do you mean #278 issue?

**Comment by jtamames** [comment]
Yes, sorry, #278

**Comment by chandnisidhu** [comment]
okay, it means removing extra "+_" from FragGeneScan files should work on this issue.

**Comment by chandnisidhu** [comment]
Hi,

I restarted the pipeline with modified gff/fna/faa files as you suggested. After step8 blastx, results from 08_gff file are like this:

`contig_13	Diamond Blastx	CDS	1	534	?	+	?	ID=contig_13_1-534;
contig_13	FGS	CDS	1	534	.	+	0	ID=contig_13_1_534;product=predicted protein
contig_13	Diamond Blastx	CDS	537	1187	?	+	?	ID=contig_13_537-1187;
contig_13	FGS	CDS	537	1187	.	-	2	ID=contig_13_537_1187;product=predicted protein
contig_13	Diamond Blastx	CDS	1281	1550	?	+	?	ID=contig_13_1281-1550;
contig_13	FGS	CDS	1281	1550	.	-	2	ID=contig_13_1281_1550;product=predicted protein
contig_13	Diamond Blastx	CDS	1707	2198	?	+	?	ID=contig_13_1707-2198;
contig_13	FGS	CDS	1707	2198	.	-	2	ID=contig_13_1707_2198;product=predicted protein
contig_13	FGS	CDS	2216	2433	.	+	1	ID=contig_13_2216_2433;product=predicted protein
contig_13	FGS	CDS	2553	2826	.	-	2	ID=contig_13_2553_2826;product=predicted protein
contig_13	Diamond Blastx	CDS	3255	3410	?	+	?	ID=contig_13_3255-3410;
contig_13	FGS	CDS	3255	3410	.	+	2	ID=contig_13_3255_3410;product=predicted protein
contig_13	Diamond Blastx	CDS	3495	3770	?	+	?	ID=contig_13_3495-3770;
contig_13	FGS	CDS	3495	3770	.	+	2	ID=contig_13_3495_3770;product=predicted protein
contig_13	Diamond Blastx	CDS	3960	10880	?	+	?	ID=contig_13_3960-10880;
contig_13	Diamond Blastx	CDS	10841	11845	?	+	?	ID=contig_13_10841-11845;
contig_13	FGS	CDS	3960	10923	.	+	2	ID=contig_13_3960_10923;product=predicted protein
contig_13	FGS	CDS	10925	11855	.	+	1	ID=contig_13_10925_11855;product=predicted protein
contig_13	Diamond Blastx	CDS	11878	12354	?	+	?	ID=contig_13_11878-12354;
contig_13	Diamond Blastx	CDS	12356	12682	?	+	?	ID=contig_13_12356-12682;
contig_13	Diamond Blastx	CDS	12687	13010	?	+	?	ID=contig_13_12687-13010;
contig_13	Diamond Blastx	CDS	13031	13327	?	+	?	ID=contig_13_13031-13327;
contig_13	FGS	CDS	11878	13327	.	+	0	ID=contig_13_11878_13327;product=predicted protein`

So basically Blastx also identified the ORFs already identified by FGS rather than just finding the new ones. Would this affect the final tables or abundance values?

Best Regards,

Chandni

**Comment by jtamames** [comment]
Hello
If the predicted ORF was found to be a correct gene (one with valid diamond hits, meaning that it likely corresponds to a real gene), then the original gene prediction is kept. Blastx prevails only in cases in which there is no gene prediction or the predicted gene does not match anything. This is explained in the manual.
Best,
J

**Comment by jtamames** [comment]
Indeed. There are overlapping entries that should not be there. Could I see your syslog please?

**Comment by chandnisidhu** [comment]
Yeah. You can find it attached. It is a restarted project, so go to line 1505.

[syslog.txt](https://github.com/jtamames/SqueezeMeta/files/6323718/syslog.txt)

**Comment by jtamames** [comment]
This points again to same naming issue. Could you please send me the 03*gff file (in intermediate), the 08*gff file (in results), and the 08*tax.wranks file (in results)? As these are probably big files, is it enough with the first 50 lines or so.
Thank you!

**Comment by chandnisidhu** [comment]
Kkay. Please find them attached. All file contain first 100 lines.

[SQM_short_files.tar.gz](https://github.com/jtamames/SqueezeMeta/files/6325885/SQM_short_files.tar.gz)

**Comment by jtamames** [comment]
Ok, got it. It's again the FGS names in the gff file. I told you to remove the final `_+` and `_-`, but I didn't realize that gene positions are separated by `_` instead of `-`. I mean, your current annotations in the 03 file are like:
`ID=contig_10000_1_537`
and they should be:
`ID=contig_10000_1-537`
If you change these accordingly in the 03.SQM_2020_MT.gff and 03.SQM_2020_MT.faa, and restart from step 4, it should work. 
Best,
Javier

**Comment by chandnisidhu** [comment]
Oh..Thanx a lot :)

**Comment by chandnisidhu** [comment]
It worked. Thanks :)

**Comment by chandnisidhu** [comment]
> Ok, got it. It's again the FGS names in the gff file. I told you to remove the final `_+` and `_-`, but I didn't realize that gene positions are separated by `_` instead of `-`. I mean, your current annotations in the 03 file are like:
> `ID=contig_10000_1_537`
> and they should be:
> `ID=contig_10000_1-537`
> If you change these accordingly in the 03.SQM_2020_MT.gff and 03.SQM_2020_MT.faa, and restart from step 4, it should work.
> Best,
> Javier

Hello,

I read the SQM 1.4 guide pdf, it says to use "_" instead of "-" in gff file to separate contig names and orfs for sqm_mapper.pl. It is true for default pipeline also when we use external gff files? Because I previously did this (as mentioned above) and it didn't work at that time.

Best,

Chandni

**Comment by jtamames** [comment]
Oh yes! My mistake, that part of the manual was ambiguous. The correct format of gene names in all the SqueezeMeta pupeline is:
`the id being the contigname and, separated by “_”, the initial and final positions of the gene (and these separated by “-”)`

Something like “contig1_1-580”

I have corrected this in the manual. Thanks for pointing this!
Now, you say that it does not work for you using this format?

Best,
J

**Comment by chandnisidhu** [comment]
No no, I worked well.
I meant it didn't work with "_".

**Comment by chandnisidhu** [comment]
Actually, I am facing some issue again with the pipeline (#error 298), So I was reading the guide pdf again and I came across this..so I thought it might be gff file issue but it is not. It is something else. I am trying to figure it out but all files seem normal, the only difference is that my current project has external bins included.

**Comment by jtamames** [comment]
Is your error related to `sqm2tables.py`? If it is, I messed it up and broke it accidentally in the latest release. Could you check that your script is functional, or download it again and try)

**Comment by fpusan** [comment]
Nope, it also fails with the corrected script. See my answer in #298.

**Comment by chandnisidhu** [comment]
Script is working fine with other projects.
